### ISSUE TITLE
Promise support!

### ISSUE THREAD

**Issue by Rob--** [chill117/express-mysql-session#67]
Hi,

Thanks for the great library. I'm using [node-mysql2](https://github.com/sidorares/node-mysql2) because it is a lot faster and because it has promise support. The problem is this library is not compatible with queries in the form of promises. Right now it only has support for queries that invoke callbacks therefore making it incompatible with `node-mysql2` promise-based connections.

This is a really easy fix, it could either detect if the connection passed into the library is promise based or an option could be based in that flags the connection as promise based.

This would only require putting any `connection.query` calls inside of a wrapper so it can make the query via callback or promise based on the options passed in.

e.g:

```javascript
MySQLStore.prototype._query = function (sql, params, callback) {
  if (this.options.promise) {
    this.connection.query(sql, params)
      .then((rows) => {
        callback(null, rows);
      }).catch((error) => {
        callback(error, null);
      });
  } else {
    this.connection.query(sql, params, callback);
  }
}
```

Note that I have just typed the above code out, it's not tested and is just a basis for the suggested implementation. All current `this.connection.query` calls would then be replaced with `this._query`.

Hopefully this all makes sense, if you have any questions feel free to ask.

**Comment by chill117** [comment]
If you have the time to put together a PR for this, I will review and merge it when I can.

**Comment by aljones15** [comment]
I got it working by modifying the connectPromise actually:

 18 const db = conObj.connectionPromise;
 21 db.query = function (sql, params, callback) {
 22   this.then((db) => db.query(sql, params))
 23     .then(([results, model]) => { 
 24       callback(null, results);
 25     }).catch((error) => {
 26       callback(error, null);
 27     });
 28 } 

 29 const MySQLStore = sqlSessionStore(session);
 30 
 31 const sessionStore = new MySQLStore({}, db);

so far no issues

**Comment by chill117** [comment]
Interesting. Nice work! And thanks for the update :)

**Comment by aljones15** [comment]
Well I left it here for anyone else in need. Has this been patched yet? I'm not sure it's really wuite production auality, but I can contribute if you want.

**Comment by chill117** [comment]
I will have a go at adding promise support, but I am not sure when I will have time.

**Comment by aljones15** [comment]
If I can make mode look reasonable then I will release an adaptor then or something. No rush man our project is working fine with just one simple addition.

**Comment by wouters-frederik** [comment]
I gave it a try with await async 
Your feedback is more than welcome.
https://github.com/chill117/express-mysql-session/pull/79

**Comment by aljones15** [comment]
has this been solved? I created an npm adapter for the issue here:

https://github.com/aljones15/express-sql-session-promise-adapter

I noticed the pull request so I can remove my project if newer version express-mysql-session work with promises.

**Comment by smulder** [comment]
Any luck? I upgraded mysql2 to get some stability fixes and started using their promise integration... then boom, express-mysql-session blew up. Is this going to be fixed anytime soon? Thanks for the awesome library, just have a site I was just about to launch and am a little worried.

Scott

**Comment by mysuf** [comment]
I don't think so. Methods don't return Promise (neither native nor Bluebird) as expected.
